truss/base/trt_llm_config.py

```diff
@@ -447,67 +459,118 @@ class ImageVersions(PydanticTrTBaseModel):
     # backend defaults and `ImageVersionsOverrides` from the pushed config.
     bei_image: str
     briton_image: str
+    v2_llm_image: str = "no-set-image"  # TODO: remove default once billip is up.
 
 
 class TRTLLMConfiguration(PydanticTrTBaseModel):
     build: TrussTRTLLMBuildConfiguration
+    inference_stack: InferenceStack = InferenceStack.v1
     runtime: TrussTRTLLMRuntimeConfiguration = TrussTRTLLMRuntimeConfiguration()
+    runtime_v2: Optional[V2TRTLLMRuntimeConfiguration] = None
+
     # If versions are not set, the baseten backend will insert current defaults.
     version_overrides: VersionsOverrides = VersionsOverrides()
 
-    def model_post_init(self, __context):
-        self.add_bei_default_route()
-        self.chunked_context_fix()
+    @model_validator(mode="after")
+    def validate_inference_stack_v2(self: "TRTLLMConfiguration", context):
+        """Validate that the build configuration is compatible with the v2 inference stack."""
+        if self.inference_stack != InferenceStack.v2:
+            return self
 
-    def chunked_context_fix(self: "TRTLLMConfiguration") -> "TRTLLMConfiguration":
-        """check if there is an error wrt. runtime.enable_chunked_context"""
-        if (
-            self.runtime.enable_chunked_context
-            and (self.build.base_model != TrussTRTLLMModel.ENCODER)
-            and not (
-                self.build.plugin_configuration.use_paged_context_fmha
-                and self.build.plugin_configuration.paged_kv_cache
-            )
-        ):
-            raise ValueError(
-                "If trt_llm.runtime.enable_chunked_context is True, then trt_llm.build.plugin_configuration.use_paged_context_fmha and trt_llm.build.plugin_configuration.paged_kv_cache need to be True. "
-            )
+        allowed_modify_fields = [
+            "checkpoint_repository",
+            "quantization_type",
+            "quantization_config",
+        ]
+
+        build_settings = self.build.model_dump(exclude_unset=True)
+        build_settings_reference = TrussTRTLLMBuildConfiguration(
+            checkpoint_repository=CheckpointRepository(
+                source=CheckpointSource.HF, repo="michael/any", revision=None
+            ),
+            quantization_type=TrussTRTLLMQuantizationType.NO_QUANT,
+            quantization_config=TrussTRTQuantizationConfiguration(),
+        ).model_dump(exclude_unset=False)
+        for field in build_settings:
+            if (
+                field not in allowed_modify_fields
+                and build_settings[field] != build_settings_reference[field]
+            ):
+                raise ValueError(
+                    f"Field trt_llm.build.{field} is not allowed to be set when using v2 inference stack. "
```

Comment:
How would a user specify quantization here? Is it simply inferred from the model weights?
If so, we are removing support for on-the-fly quantization?